enzyme/Enzyme/PreserveNVVM.cpp

```diff
@@ -169,12 +169,135 @@ class PreserveNVVM final : public FunctionPass {
           if (auto GV = cast<GlobalVariable>(V)) {
             GV->setMetadata("enzyme_inactive", MDNode::get(g.getContext(), {}));
             toErase.push_back(&g);
+            changed = true;
+          } else {
+            llvm::errs() << "Param of __enzyme_inactive_global must be a "
+                            "global variable"
+                         << g << "\n"
+                         << *V << "\n";
+            llvm_unreachable("__enzyme_inactivefn");
+          }
+        }
+      }
+      if (g.getName().contains("__enzyme_inactivefn")) {
+        if (g.hasInitializer()) {
+          Value *V = g.getInitializer();
+          while (1) {
+            if (auto CE = dyn_cast<ConstantExpr>(V)) {
+              V = CE->getOperand(0);
+              continue;
+            }
+            if (auto CA = dyn_cast<ConstantAggregate>(V)) {
+              V = CA->getOperand(0);
+              continue;
+            }
+            break;
+          }
+          if (auto F = cast<Function>(V)) {
+            F->addAttribute(AttributeList::FunctionIndex,
```

Comment:
Do you mean "getAttributes" here instead of "addAttribute"? This is what causes the error in the MLIR build

```bash
/home/ubuntu/actions-runner/_work/Enzyme/Enzyme/enzyme/Enzyme/PreserveNVVM.cpp:197:16: error: ‘class llvm::Function’ has no member named ‘addAttribute’; did you mean ‘getAttributes’?
```